lib/auth/webauthncli/api.go

```diff
@@ -92,11 +92,16 @@ func Login(
 	ctx context.Context,
 	origin string, assertion *wanlib.CredentialAssertion, prompt LoginPrompt, opts *LoginOpts,
 ) (*proto.MFAAuthenticateResponse, string, error) {
+	if origin == "" {
+		return nil, "", trace.BadParameter("origin required")
+	}
+	if err := assertion.Validate(); err != nil {
+		return nil, "", trace.Wrap(err)
+	}
```

Comment:
I would push this further down into fido2Login, fido2Register, `touchid` and `webauthnwin` package.

As a rule of thumb, validation is better done where the data is used, not in intermediate layers. The methods on `touchid` and `webauthnwin` are public, if someone calls them directly they should still make sure the data is good.